src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -1,15 +1,127 @@
+  title?: string;
+  unit_of_measurement?: string;
+  min?: number;
+  max?: number;
+  severity?: object;
+}
+
+class HuiGaugeCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error("Invalid card configuration");
+    }
+    if (!isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+    this._config = { min: 0, max: 100, ...config };
+  }
+
+  protected render() {
+    if (!this._config) {
+      return html``;
+    }
+    const stateObj = this.hass!.states[this._config.entity];
 
-/*
- * @appliesMixin EventsMixin
- */
-class HuiGaugeCard extends EventsMixin(PolymerElement) {
-  static get template() {
+    return html`
+    ${this.renderStyle()}
+    <ha-card  @click="${this._handleClick}">
+    ${
+      !stateObj
+        ? html`<div class="not-found">Entity not available: ${
+            this._config.entity
+          }</div>`
+        : html`
+          <div class='container'>
+            <div class='gauge-a'></div>
+            <div class='gauge-b'></div>
+            <div class='gauge-c' id='gauge'></div>
+            <div class='gauge-data'>
+              <div id='percent'>${stateObj.state}
+              ${this._config!.unit_of_measurement ||
+                stateObj.attributes.unit_of_measurement ||
+                ""}
+              </div>
+              <div id='title'>${this._config.title}
+
+              </div>
+            </div>
+          </div>
+        `
+    }
+    </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
+    if (changedProps.get("hass")) {
+      return (
+        changedProps.get("hass").states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
+      );
+    }
+    return changedProps;
+  }
+
+  protected firstUpdated() {
+    const stateObj = this.hass!.states[this._config!.entity];
+
+    if (isNaN(Number(stateObj.state))) {
+      this.shadowRoot!.querySelector("title")!.classList.add("not-found");
+      this.shadowRoot!.querySelector("title")!.innerHTML =
+        "Entity is non-numeric: " + this._config!.entity;
+    }
+  }
+
+  protected updated() {
+    const stateObj = this.hass!.states[this._config!.entity];
+    const turn = this._translateTurn(stateObj.state, this._config);
+
+    this.shadowRoot!.getElementById(
+      "gauge"
+    )!.style.transform = `rotate(${turn}turn)`;
+
+    this.shadowRoot!.getElementById(
+      "gauge"
+    )!.style.backgroundColor = this._computeSeverity(
+      stateObj.state,
+      this._config!.severity
+    );
+
+    if (this.clientWidth < 200) {
+      (this.shadowRoot!.querySelector(
+        "ha-card"
+      )! as HTMLElement).style.setProperty(
+        "--base-unit",
```

Comment:
I prefer to have us toggle classes than to mutate CSS directly, that way it will make it possible for people to override the CSS by just overriding `renderStyle`